### PR TITLE
Check max HTLC when forwarding payments.

### LINKED ISSUE #2182: lnwire+funding+routing: populate, implement and adhere to the new max_htlc ChannelUpdate value
Recently, a new `max_htlc` value has been added to the ChannelUpdate message via a spec amendment: https://github.com/lightningnetwork/lightning-rfc/pull/465. This value in the short term, can allow nodes to signal the largest HTLC they're willing to carry, allow light clients who don' t verify channel existence to have some guidance when routing HTLCs, and finally may allow nodes to preserve a portion of their available bandwidth at all times by advertising a value lower than the actual channel capacity. In the long term, this can be used by implementations of AMP to guide payment splitting, as it becomes apparent to a node the _largest_ possible HTLC (in an idea world of full bandwidth available for each link direction one wishes to route) one can route over a particular channel. 

As a prep for our AMP implementation, and also to provide those existing naive light clients with additional path finding implementation, we should add and observe this new field. Now that PR #1825 has been merged, older nodes will properly retain and verify signatures including these new fields even though they haven't yet updated to recognize these fields. As a result, only now can we proceed with out implementation. 

# Steps To Completion 

- [ ] Update the `ChannelUpdate` struct in `lnwire` to partition the existing flags field into the new `channel_flags` and `message_flags` pairs. 

- [ ] Add a new `MaxHTLC` field to the `ChannelUpdate` message.

- [ ] When funding new channels, set a default for these values if one isn't specified. 

- [ ] Extend the `UpdateChanPolicy` RPC and cli commands to allow the caller to specify this value. This may require modifications to the peer and switch as well to ensure these fields properly propagate and are set properly. 

- [ ] Add new verification logic within the link [`HtlcSatifiesPolicy`](https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L1874) method to ensure the HTLC isn't too large. 

- [ ] Update [the path finding logic](https://github.com/lightningnetwork/lnd/blob/master/routing/pathfind.go#L545) to ignore a channel if the HTLC value (including the fees at that point) exceed the max HTLC value (if set) of the link.

### PR DESCRIPTION
**Pull request by valentinewallace**
Depends on #2434 (which depends on #2296). Partially closes #2182.

In our previous PR, we implement parsing, storing and re-sending the new max HTLC field that has been added to ChannelUpdates.

In this PR, we verify an HTLC does not exceed the max HTLC for a given link before forwarding it.

### CODE REVIEW

**Comment by Roasbeef** [comment]
Needs a rebase.

**Comment by valentinewallace** [comment]
Updated [pending travis]! @Roasbeef

File: htlcswitch/link_test.go [hunk 1]
```diff
@@ -770,6 +770,69 @@ func TestLinkForwardMinHTLCPolicyMismatch(t *testing.T) {
...
 
+// TestLinkForwardMaxHTLCPolicyMismatch tests that if a node is an intermediate
+// node and receives an HTLC which is _above_ its max HTLC policy then the
+// HTLC will be rejected.
+func TestLinkForwardMaxHTLCPolicyMismatch(t *testing.T) {
+	t.Parallel()
+
+	channels, cleanUp, _, err := createClusterChannels(
+		btcutil.SatoshiPerBitcoin*5,
+		btcutil.SatoshiPerBitcoin*5)
```
Comment (wpaulino) [inline htlcswitch/link_test.go]: Nit: Args should be on their own lines if they can't fit, e.g.:
```
createClusterChannels(
    btcutil.SatoshiPerBitcoin*5, btcutil.SatoshiPerBitcoin*5,
)
```
There are a couple others like this throughout the commit.

Comment (valentinewallace) [reply htlcswitch/link_test.go]: Fixed, although the commit has moved to #2434. Closing this PR now :)

**Pull request closed without merging**